### ISSUE TITLE
Request your support to use this code

### ISSUE THREAD

**Issue by vnchandan** [yaqiz01/cs231n#1]
Hi,
We want to replicate below result same as mentioned in paper.
"Table 1 shows the train, validation, and test MSE of our
best model, with the baseline architecture trained on images
averaged and downsampled to 100x300 and optical flow and
object detection masks as input"

We have downloaded dataset using download_data.py.

We are using below command to train the model.
python play.py --mode=trainspeed --model=conv --speedmode=1 --convmode=0
Result: Epoch 15, Overall train mse = 1.557, Overall val mse = 6.85

We are testing using below command. Note: commented lightdet & signdet in python code play() function.
python play.py --mode=all --model=conv --speedmode=1 --convmode=0
Result:
Overall averaged test mse wu: 3.07688334064
Overall averaged test mse vf: 16.8498900062
Overall averaged test mse af: 1.04141423059

Please help.
Thanks

**Comment by yaqiz01** [comment]
Hi,

Thanks for the inquiry. Unfortunately I couldn't remember which exact configuration produced the result that we had in the paper since it was a while back since we did the class project. But all our results and their configurations are in the results/ folder and the better results are with later time stamps. I found a couple rans that had similar training and validation result. for example 
`result_20170613022031.txt`
The configuration is printed at the beginning of the file. 
For testing, you probably want to use `testspeed` mode instead of `all` mode to make sure we are using the same testpath/ test video.

**Comment by vnchandan** [comment]
Hi,

We need a clarification on one of the thing related to vehicle motion
detection model.
We are trying motion detection using optical flow only.
As per the paper and git hub code, sampling every 3rd frame for training is
done.
Meaning your input are images captured at ~3fps.

But while testing,  10fps kitti dataset video is directly used.
We feel that output like velocity should be 3times smaller, as the Optical
flow captured will be less in 10fps compared to 3fps.

Can you provide some insight on above?

Thanks,
Chandan

**Comment by yaqiz01** [comment]
I'm not sure I entirely follow the question. If you look at options in play.py, there's a sample_every option which sample one frame every three frames from the dataset. However,  that does not mean our sampling rate is 10 frame per second, our sampling rate will be sampling rate of the original kitti dataset divide by 3. During testing the same option is used, i.e. we only take one frame every three frames from the dataset.

**Comment by vnchandan** [comment]
Hi,

We are confused due to the code.
In github latest Play.py line:111 should be changes to below?
    if mode in ['trainspeed', 'testspeed']

Thanks,
Chandan

**Comment by yaqiz01** [comment]
Ah that does seem to be a bug to me. I pushed the fix you suggested, and also a demo video. You can see that our predicted speed is indeed faster than the ground truth in general, but not by 3x. I'm not sure why, though. From paper our test mse is much higher and I suppose it should be much smaller. The validation set uses the same setting as training so shouldn't be effected.
